todo-list/src/components/todos/todoList.tsx

```diff
@@ -0,0 +1,59 @@
+import { useAddTodo, useTodos } from "@/hooks/todos/useTodo";
+import { useState } from "react";
+import { Input } from "../ui/input";
+import { Button } from "../ui/button";
+import { TodoItem } from "./todoItem";
+import { Todo } from "@/types/todos/todos.type";
+
+export function TodoList() {
+  const { todos, isLoading, isError } = useTodos();
+  const { handleAddTodo, isLoading: isAdding } = useAddTodo();
+  const [newTodo, setNewTodo] = useState("");
+
+  const addNewTodo = () => {
+    if (newTodo.trim()) {
+      handleAddTodo(newTodo, false);
+      setNewTodo("");
+    }
+  };

```

Comment:
should add await to handleAddTodo

Reply:
Read more about query invalidation

https://tanstack.com/query/v4/docs/framework/react/guides/query-invalidation